examples/plot_clip.py

```diff
@@ -0,0 +1,125 @@
+"""
+Clip Vector Data with EarthPy
+==================================================================
+
+Learn how to clip point, line, or polygon geometries with a
+polygon geometry using EarthPy.
+
+"""
+
+###############################################################################
+# Clip Vector Data in Python Using EarthPy
+# ---------------------------------------------
+#
+# .. note::
+#    The example below will show you how to use the clip_shp() function to clip
+#    vector data such as points, lines and polygons to a vector boundary.
+#
+# The example below walks you through a typical workflow for clipping one
+# vector data file to the shape of another. Both vector data files must be
+# opened with GeoPandas as GeoDataFrames and be in the same Coordinate
+# Reference System (CRS) for the ``clip_shp()`` function in EarthPy to work.
+#
+# This example uses a line shapefile containing all major roads in North America
+# and a polygon shapefile containing the United States boundary to show you how
+# to prepare and clip vector data. In this example, the roads will be clipped
+# to the United States boundary.
+#
+# .. note::
+#    The object to be clipped will be clipped to the full extent of the clip
+#    object. If there are multiple polygons in clip object, data from shp will
+#    be clipped to the total boundary of all polygons in clip object.
+
+###############################################################################
+# Import Packages
+# ---------------
+#
+# To begin, import the needed packages. You will primarily use EarthPy's clip
+# utility alongside GeoPandas.
+
+import os
+import matplotlib.pyplot as plt
+import geopandas as gpd
+import earthpy as et
+import earthpy.clip as ec
+
+###############################################################################
+# Import Example Data
+# -------------------
+#
+# Once the packages have been imported, download the data needed for this
+# example: one line shapefile containing all major roads in North America,
+# and one polygon shapefile containing the United States boundary.
+
+data = et.data.get_data("spatial-vector-lidar")
+
+###############################################################################
+# Open Files with GeoPandas and Reproject the Data
+# -------------------------------------------------
+#
+# Start by setting your working directory. Then, import the data files to
+# GeoDataFrames using GeoPandas.
+#
+# Recall that the data must be in the same CRS in order to use the
+# ``clip_shp()`` function. If the data are not in the same CRS, be sure to use
+# the ``to_crs()`` function from GeoPandas to match the projects between the
+# two objects, as shown below.
+
+# Set your home environment
+os.chdir(os.path.join(et.io.HOME, "earth-analytics"))
+
+# Open both files with GeoPandas
+roads = gpd.read_file(
+    os.path.join(
```

Comment:
In our previous vignettes, we have been using relative paths such as:
```
country_boundary = gpd.read_file(
    "data/spatial-vector-lidar/usa/usa-boundary-dissolved.shp"
```
I believe @nkorinek changed the data path to use `os.path.join()` because in my review, I asked him to remove an extra ```r``` he included in front of the relative path, which is needed only for Windows.

@lwasser What would you like implemented here?

Reply:
i think os.path.join is reasonable here. i suggest rather than embed it within the read_file function, why now define the path and then read it in like:

```
us_bound_path = os.path.join(code-here)
read_file(us_bound_path)
```
that is the most reproducible way to do it i think with the least amount of nesting